### PR TITLE
Stop test execution early for changes to docs only

### PR DESCRIPTION
**Pull request by webvictim**
This PR uses `git diff --raw` to check whether there have been any changes to the code other which don't match `^docs/` or `*.md`. If there haven't, it will tell Drone to skip unit tests/integration tests to save CI time for docs-only PRs.

This will require me to watch PRs closely for a while to make sure that the tests are only ever skipped for docs PRs.

This is mostly a proof of concept for optionally skipping pipelines in Drone. It may not be worth the risk given that our CI pipeline is down to just under 10 minutes on average now. I'm mostly just proud of the fact that I helped the Drone authors to find and fix a bug to make this work (https://discourse.drone.io/t/using-exit-code-78-with-kubernetes-runner-doesnt-stop-execution/7937/8)

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -74,6 +74,29 @@ steps:
...
+  # which don't match the pattern '^docs/' or '.md$' were changed. if there are no
+  # changes to non-docs code, we skip the Teleport tests and exit early with a special
+  # Drone exit code to speed up iteration on docs (as milv is much quicker to run)
+  - name: Optionally skip tests
+    image: docker:git
+    commands:
+    - |
+        cd /go/src/github.com/gravitational/teleport
+        echo -e "\n---> git diff --raw ${DRONE_COMMIT}..origin/${DRONE_COMMIT_BRANCH:-master}\n"
+        git diff --raw ${DRONE_COMMIT}..origin/${DRONE_COMMIT_BRANCH:-master}
```
Comment (awly) [inline .drone.yml:88]:
```suggestion
```

**Review by awly — COMMENTED** [review]
We should at least run `make lint-sh` on docs PRs (it's included in `make lint`)

File: .drone.yml (hunk 1 above)
Comment (webvictim) [reply .drone.yml:88]: This line is here just to show the output of the command for reference so you know what is being evaluated by the line below. It's perhaps a bit hacky but it's useful to be able to see exactly what was changed in a PR (so we know whether we're running tests when we shouldn't be or, heaven forbid, vice versa)

**Comment by webvictim** [comment]
Sure, I'll add `make lint-sh`. I did actually consider always running `make lint` as well because it only takes ~1 minute - it'll still give a decently fast feedback cycle and it's pretty deterministic - I've never seen it fail without the code actually being broken.

File: .drone.yml (hunk 1 above)
Comment (webvictim) [reply .drone.yml]: Here's some example output from the step so you can see what I mean:

```console
+ cd /go/src/github.com/gravitational/teleport\necho -e "\n---> git diff --raw 86dc48a8cbe39db9fc475a8200fc8811a8bc9655..origin/master\n"\ngit diff --raw 86dc48a8cbe39db9fc475a8200fc8811a8bc9655..origin/master\ngit diff --raw 86dc48a8cbe39db9fc475a8200fc8811a8bc9655..origin/master \| awk '{print $6}' \| grep -Ev '^docs/' \| grep -Ev '.md$' \| grep -v ^$ \| wc -l > /tmp/.change_count.txt\nexport CHANGE_COUNT=$(cat /tmp/.change_count.txt \| tr -d '\n')\necho -e "\n---> Changes to Teleport code detected: $CHANGE_COUNT"\nif [ $CHANGE_COUNT -gt 0 ]; then\n  echo "---> Teleport tests will run normally"\nelse\n  echo "---> Skipping Teleport tests and exiting early"\n  exit 78\nfi\necho ""\n

---> git diff --raw 86dc48a8cbe39db9fc475a8200fc8811a8bc9655..origin/master

100644 100644 c913486d 753f9551 M	.drone.yml

---> Changes to Teleport code detected: 1
---> Teleport tests will run normally
```

**Comment by webvictim** [comment]
Updated to always run `make lint` before skipping. Total runtime is still sub-2 minutes which I think is fine. It's not quite the ~40 seconds I was managing before but it'll do.

**Review by awly — APPROVED** [review]

**Pull request merged**